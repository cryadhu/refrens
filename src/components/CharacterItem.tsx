import React from "react";
import { View, StyleSheet, ImageBackground } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";

import { Characters } from "../types/response/characters";
import { ButtonProps } from "../types/components/button";

import colors from "../utils/color";
import TextView from "./TextView";

const styles = StyleSheet.create({
  root: {
    flex: 1,
    margin: 5,
    width: "100%",
  },
  image: {
    flex: 1,
    height: 200,
    justifyContent: "flex-end",
  },
  infoContainer: {
    backgroundColor: "rgba(0,0,0,0.5)",
    paddingVertical: 10,
    paddingHorizontal: 10,
  },
  name: {
    fontFamily: "InterBold",
    color: colors.white,
  },
});

const CharacterItem = (
  props: {
    data: Characters;
  } & ButtonProps
) => {
  const { data, onPress } = props;
  const { name, image, species, status } = data;

  const onItemClick = () => {
    if (typeof onPress === "function") {
      onPress(data);
    }
  };

  return (
    <TouchableOpacity onPress={onItemClick} containerStyle={styles.root}>
      <ImageBackground source={{ uri: image }} style={styles.image}>
        <View style={styles.infoContainer}>
          <TextView numberOfLines={1} style={styles.name}>
            {name}
          </TextView>
          <TextView>{`${status}-${species}`}</TextView>
        </View>
      </ImageBackground>
    </TouchableOpacity>
  );
};

export default CharacterItem;
