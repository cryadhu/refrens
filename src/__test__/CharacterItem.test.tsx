import React from "react";
import { render, RenderAPI } from "@testing-library/react-native";
import CharacterItem from "../components/CharacterItem";

let component: RenderAPI;

function initializeComponent() {
  return render(
    <CharacterItem
      data={{
        id: 1,
        name: "test",
        image: "sample_url",
        species: "alien",
        status: "alive",
        gender: "male",
        origin: { name: "earth" },
        location: {
          name: "area-51",
        },
        episode: [],
      }}
    />
  );
}

beforeAll(async () => {
  component = await initializeComponent();
});

test("Renders correctly", async () => {
  await expect(component).toMatchSnapshot();
});
