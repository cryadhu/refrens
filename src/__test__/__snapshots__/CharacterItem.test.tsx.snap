// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Object {
      "flex": 1,
      "margin": 5,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
          "height": 200,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "sample_url",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 200,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#ffff",
              "fontFamily": "InterBold",
            }
          }
        >
          test
        </Text>
        <Text
          style={
            Object {
              "color": "#ffff",
              "fontFamily": "InterRegular",
            }
          }
        >
          alive-alien
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
